pkg/providers/tinkerbell/create.go

```diff
@@ -42,50 +43,84 @@ func (p *Provider) BootstrapClusterOpts() ([]bootstrapper.BootstrapClusterOption
 }
 
 func (p *Provider) PreCAPIInstallOnBootstrap(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error {
-	if p.setupTinkerbell {
-		logger.V(4).Info("Installing Tinkerbell stack on the bootstrap cluster")
-		if err := p.InstallTinkerbellStack(ctx, cluster, clusterSpec, true); err != nil {
-			return fmt.Errorf("installing tinkerbell stack on the bootstrap cluster: %v", err)
-		}
+	if !p.setupTinkerbell {
+		return nil
+	}
+
+	logger.V(4).Info("Installing Tinkerbell stack on bootstrap cluster")
+
+	ip, err := networkutils.GetLocalIP(p.netClient)
+	if err != nil {
+		return err
+	}
+
+	stack := newStack(clusterSpec.VersionsBundle.Tinkerbell.TinkerbellStack, p.docker, p.writer, p.helm, ip.String()).
+		withNamespace(constants.EksaSystemNamespace, false).
+		withoutBoots().
+		withHegel().
+		withTinkController().
+		withTinkServer()
+
+	if err := stack.install(ctx, cluster.KubeconfigFile); err != nil {
+		return fmt.Errorf("install Tinkerbell stack on bootstrap cluster: %v", err)
+	}
+
+	if err := stack.installBootsOnDocker(ctx, cluster.KubeconfigFile); err != nil {
+		return err
```

Comment:
I think there's too much implementation detail here. Why does the provider care about the fact boots is installed on docker? Can we model a default installer's behavior, which includes all components of the stack, and offer options where control is needed (probably only boots?)? 

For example:

```go
installer := NewStackInstaller(..., InstallBootsOnDocker())

if err := installer.Install(ctx, kubeconfig); err != nil {
  return ...
}
```

Here, the provider only cares about 2 things: (1) create an installer; (2) run installation. This paves the way to extract the installer and inject it having the provider leverage an interface scoped to `Install` only (_we don't need to do this now_).

```go
type StackInstaller interface {
  InstallStack(context.Context, string) error
}
```

I mention this because its also a little odd that the provider consumes a `Helm` related object. The provider should care what mechanism is used to install anything - only that it controls _when_ the install cmd happens.